### PR TITLE
Basic from_bytes expose to external cython modules and 

### PR DESCRIPTION
**Pull request by danielfrg**
Some context: I am building a cython wrapper to [libhdfs3](https://github.com/PivotalRD/libhdfs3) ([cyhdfs3](https://github.com/danielfrg/cyhdfs3)) to read data from HDFS more natively on python and I wanted to be able to read avro files that are on HDFS.

The original plan was to pass the bytes from python to `cyavro` (adding a simple `from_bytes` classmethod) but data gets mysteriously changed when passing the python string back to cython so I decided to pass the data directly from libhdfs3 to libavro using cython, that actually makes me feel better.

So I created a `_cyavro.pyd` that other cython modules can `cimport cyavro`: https://github.com/danielfrg/cyhdfs3/blob/cyavro/cyhdfs3/cyhdfs3.pyx#L7

I also added a basic `cdef from_bytes` to `_cyavro` that uses `fmemopen` so I could basically reuse the whole library by using just `avro_file_reader_fp` and setting the `AvroReader._reader` attribute.

At the moment this requires `posix.stdio` which is on the cython repo now (https://github.com/cython/cython/blob/master/Cython/Includes/posix/stdio.pxd) but its not in any released versions so i copy pasted that small file for now.

Everything is working on my simple tests I just had to remove the reader reset at the end of `init_buffer`, is this necessary for reading a localfile? In my test it didn't look like i need it.

Is there some interest on this PR? I can order the code in other ways that you find better (my first time using cython). I could probably just need `_cython.pxd` but maybe its nice to have the `from_bytes` in this library.

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
Can you add a test for this please?

Looks good to me.  This would be very handy stuff in conjunction with some of the hdfs libraries for python

**Comment by danielfrg** [comment]
I changed a little bit the structure of the original commits since I wanted a `classmethod` but you cannot have `cdef` classmethod so I ended up doing a `reader_from_bytes_c` function that returns an `AvroReader` as suggested in the cython mailing list.

I also changed the `from_bytes` method to a more general `init_reader_buffer` similar to `init_reader` you have.

I also finally figured out that the reset at the end of `init_buffers` I removed for my reader is because the first record is read and therefore lost when you do `read_chunch` so I also fixed that and added a small tests that saves an avro file and reads back using `reader_from_bytes_c`.

I also added the `.pyx` and `.pyd` files to the conda package since they are needed to `cimport` from other cython modules.

**Comment by mariusvniekerk** [comment]
Yeah the reset is there to prevent losing record 1.  You unfortunately have to read one record in order to build the numpy array buffers that it feeds into.  

Can you add some read_bytes_as_X methods to

`cyavro/__init__.py`

That would make it slightly easier for folks to use it in conjunction with hdfs-cli.  Figure something like a 
`read_bytes_as_dataframe`

Or i guess we could overload the read functions similar to how pandas does it for read_csv to work on a string or a file-like object and just immediately read all the bytes if passed a file like object.

**Comment by danielfrg** [comment]
That would be a nice function and my original intent but unfortunately if I do something like this in python:

``` python
# t.py
f = open('f.avro', 'rb')
data_py = f.read()

cyavro.from_bytes(data_py)

# _cyavro.pyx
def from_bytes(self, bytes data):
    cdef void *buffer = <void*> data
    # pass buffer to the reader
```

I got a lot of `Incorrect magic number...` errors, basically saying that the data is not valid avro. I tried a bunch of ways to pass the data back from python to C like memory views but nothing worked.

So my solution was to pass the `void *` directly from my `cyhdfs3` to `cyavro` that way there is no python just C. That makes it that cyhdfs3 requires cyavro and needs to be linked to `libavro` but I think I am happy with that specially if it also removes some overhead.

So basically there is no way to use the `reader_from_bytes` from python at the moment, if you have suggestions on how to do that I would be happy to test something I missed.

**Comment by mariusvniekerk** [comment]
Yeah I think I can try to tackle the python interface in a different pr.  Merging
